### PR TITLE
Add Executable Items / Block Support

### PR DESCRIPTION
**Pull request by Oribuin**
Implement ExecutableItems and ExcecutableBlocks support for deluxemenus.

ExecutableBlocks seems to have a lack of information regarding their maven dependency with the one provide being outdated and missing from repositories, Suppressing deprecation is not ideal.

### CODE REVIEW

File: src/main/java/com/extendedclip/deluxemenus/hooks/ExecutableBlocksHook.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+public class ExecutableBlocksHook implements ItemHook, SimpleCache {
+
+  private final Map<String, ItemStack> cache = new ConcurrentHashMap<>();
+
+  @SuppressWarnings("deprecation")
```
Comment (BlitzOffline) [inline src/main/java/com/extendedclip/deluxemenus/hooks/ExecutableBlocksHook.java:15]: This is indeed not that nice. After looking at a decompiled version of the plugin, it seems that it does indeed have an API and the wiki suggests using the JAR directly. No remote repository is provided.

I guess, for now we'll support the deprecated version until the developer decides to completely remove it or they provide an online version of the dependency..

File: src/main/java/com/extendedclip/deluxemenus/utils/Constants.java [hunk 2]
```diff
@@ -42,6 +42,8 @@ private Constants() {
     public static final String ITEMSADDER_PREFIX = "itemsadder-";
     public static final String ORAXEN_PREFIX = "oraxen-";
     public static final String MMOITEMS_PREFIX = "mmoitems-";
+    public static final String EXECUTABLEITEMS_PREFIX = "ei-";
```
Comment (iGabyTM) [inline src/main/java/com/extendedclip/deluxemenus/utils/Constants.java:45]: Please use the full name for the prefix, same for the other.

Comment (iGabyTM) [inline src/main/java/com/extendedclip/deluxemenus/utils/Constants.java:45]: Please use the full name for the prefix, `executableitems-` and `executableblocks-`.

**Review by iGabyTM — CHANGES_REQUESTED** [review]

**Review by BlitzOffline — CHANGES_REQUESTED** [review]

**Comment by Oribuin** [comment]
I'd like to hold this pull request off until #53 is merged so that both branches can work together.

**Review by iGabyTM — DISMISSED** [review]

**Review by Kqliber — DISMISSED** [review]

**Comment by Oribuin** [comment]
FIxed all the merge conflicts with main, everything should be ready to go.

**Comment by BlitzOffline** [comment]
There seems to still be some merge conflicts. Also, considering Gaby suggested it as well, I think it is best to use `executableitems-` and `executableblocks-`. The only hook still using short names is the HeadDatabase one for backwards compatibility.

**Comment by Oribuin** [comment]
Fixed merge conflict as it was literally just an extra space.

Changed execitems/execblocks to executableitems/executableblocks

File: src/main/java/com/extendedclip/deluxemenus/utils/Constants.java [hunk 3]
```diff
@@ -42,6 +42,8 @@ private Constants() {
     public static final String ITEMSADDER_PREFIX = "itemsadder-";
     public static final String ORAXEN_PREFIX = "oraxen-";
     public static final String MMOITEMS_PREFIX = "mmoitems-";
+    public static final String EXECUTABLEITEMS_PREFIX = "execitems-";
+    public static final String EXECUTABLEBLOCKS_PREFIX = "execblocks-";
```
Comment (BlitzOffline) [inline src/main/java/com/extendedclip/deluxemenus/utils/Constants.java:46]: Are these constants used anywhere anymore? If yes, they should be changed. If no, they should be removed.

**Review by BlitzOffline — CHANGES_REQUESTED** [review]

**Review by BlitzOffline — APPROVED** [review]

**Pull request merged**